pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -50,50 +50,50 @@ func (Plugin) OnOutboundListener(in *plugin.CallbackListenerInputParams, mutable
 	proxyInstances := in.ProxyInstances
 
 	switch in.ListenerType {
-	case plugin.ListenerTypeHTTP:
-		*mutable.HTTPFilters = append(*mutable.HTTPFilters, buildMixerHTTPFilter(env, node, proxyInstances, true))
+	case pluginPkg.ListenerTypeHTTP:
+		mutable.HTTPFilters = append(mutable.HTTPFilters, buildMixerHTTPFilter(env, node, proxyInstances, true))
 		return nil
-	case plugin.ListenerTypeTCP:
-		*mutable.TCPFilters = append(*mutable.TCPFilters, buildMixerOutboundTCPFilter(env, node))
+	case pluginPkg.ListenerTypeTCP:
+		mutable.TCPFilters = append(mutable.TCPFilters, buildMixerOutboundTCPFilter(env, node))
 		return nil
 	}
 
 	return fmt.Errorf("unknown listener type %v in mixer.OnOutboundListener", in.ListenerType)
 }
 
 // OnInboundListener implements the Callbacks interface method.
-func (Plugin) OnInboundListener(in *plugin.CallbackListenerInputParams, mutable *plugin.CallbackListenerMutableObjects) error {
+func (plugin) OnInboundListener(in *pluginPkg.CallbackListenerInputParams, mutable *pluginPkg.CallbackListenerMutableObjects) error {
 	env := in.Env
 	node := in.Node
 	proxyInstances := in.ProxyInstances
 	instance := in.ServiceInstance
 
 	switch in.ListenerType {
-	case plugin.ListenerTypeHTTP:
-		*mutable.HTTPFilters = append(*mutable.HTTPFilters, buildMixerHTTPFilter(env, node, proxyInstances, false))
+	case pluginPkg.ListenerTypeHTTP:
+		mutable.HTTPFilters = append(mutable.HTTPFilters, buildMixerHTTPFilter(env, node, proxyInstances, false))
 		return nil
-	case plugin.ListenerTypeTCP:
-		*mutable.TCPFilters = append(*mutable.TCPFilters, buildMixerInboundTCPFilter(env, node, instance))
+	case pluginPkg.ListenerTypeTCP:
+		mutable.TCPFilters = append(mutable.TCPFilters, buildMixerInboundTCPFilter(env, node, instance))
 		return nil
 	}
 
 	return fmt.Errorf("unknown listener type %v in mixer.OnOutboundListener", in.ListenerType)
 }
 
 // OnOutboundCluster implements the Callbacks interface method.
-func (Plugin) OnOutboundCluster(env model.Environment, node model.Proxy, service *model.Service, servicePort *model.Port, cluster *xdsapi.Cluster) {
+func (plugin) OnOutboundCluster(env model.Environment, node model.Proxy, service *model.Service, servicePort *model.Port, cluster *xdsapi.Cluster) {
```

Comment:
env and node should probably be ptr in the interface (for consistency if not performance). having a mix of value and ptr structs is confusing, it implies that service and port are in/out which i don't believe is the intention?

Reply:
Yeah that'd be reasonable. W.r.t. the out parameter, we should either ditch it (make the function return an `*xdsapi.Cluster`, and implementations should return `cluster`) or rename it to make the semantics more clear, calling it `out`.